SCF/SCF-1/FixedOrderPromotion.java

```diff
@@ -0,0 +1,82 @@
+public class FixedOrderPromotion implements Promotions{
+
+	private double minimumPrice;
+	private double fixedDiscount;
+	private PromotionEnum validPromo;
+	
+	public FixedOrderPromotion() {
+		// TODO Auto-generated constructor stub
+		this.setMinimumPrice(1000);
+		this.setFixedDiscount(5);
+	}
+	@Override
+	public double getMinimumPrice() {
+		return minimumPrice;
+	}
+
+	@Override
+	public void setMinimumPrice(double amount) {
+		this.minimumPrice = amount;
+		
+	}
+
+	@Override
+	public double getFixedDiscount() {
+		return fixedDiscount/100;
+	}
+
+	@Override
+	public void setFixedDiscount(double discount) {
+		this.fixedDiscount = discount;
+		
+	}
+	
+	/*
+	 * checks if the promo code entered is correct or not by looking up 
+	 * in the PromotionEnum 
+	 * @param promoCode entered by user
+	 * @returns true if it is correct else false
+	 */
+	public boolean isValidPromo(String promoCode){
+		for(PromotionEnum p : PromotionEnum.values()){
+			if(promoCode.equals(p.getCode())){

```

Comment:
Can be done using Lambda expression. It will be more optimized than for each loop.